react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -65,6 +73,29 @@ export const Reveal3DResources = ({
     setReveal3DModelsStyling(modelsStyling);
   }, [modelsStyling]);
 
+  useEffect(() => {
+    const callback = (event: PointerEventData): void => {
+      void (async (event: PointerEventData): Promise<void> => {
+        const data = await queryMappedData<NodeType>(
+          viewer,
+          client,
+          fdmSdk,
+          fdmAssetMappingConfig,
+          event
+        );
+        if (onNodeClick !== undefined && data !== undefined) {
+          onNodeClick?.(data);
+        }
+      })(event);
```

Comment:
Should consider wrapping it in a hook that uses queries but could be a different PR. Will help a little bit with performance, though.

Reply:
Agree :+1:  https://cognitedata.atlassian.net/browse/UX-1147